frontend/src/packages/api/types/filesystem.ts

```diff
@@ -0,0 +1,30 @@
+// FilesystemEntry is the contract for the type of response we receive from the backend
+export type FilesystemEntry = {
+    EntityID: string,
+    EntityName: string,
+    IsDocument: boolean,
+    Parent: string,
+    Children: FilesystemEntry[]
+}
+
+export type CreateFilesystemEntryResponse = {
+    NewID: string
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const hasFieldOfType = (o: any, fieldName: string, type: string): boolean =>
+    fieldName in o && (typeof o[fieldName]) === type;
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const IsFilesystemEntry = (o: any): o is FilesystemEntry =>
+    hasFieldOfType(o, "EntityID", "string") &&
+    hasFieldOfType(o, "EntityName", "string") &&
+    hasFieldOfType(o, "IsDocument", "boolean") &&
+    hasFieldOfType(o, "Parent", "string") && 
+    hasFieldOfType(o, "Children", typeof []) &&
```

Comment:
this isn't very reliable since `typeof []` is just `'object'` (javascript moment) - you probably want to use [`Array.isArray`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray) here